### ISSUE TITLE
`Failed to handle request: Error: Failed to set ECDH curve` on macOS

### ISSUE THREAD

**Issue by azerum** [httptoolkit/mockttp#139]
Whenever the proxy server tries to handle request, I get the following error:

```
Failed to handle request: Error: Failed to set ECDH curve
    at configSecureContext (node:internal/tls/secure-context:239:11)
    at Object.createSecureContext (node:_tls_common:117:3)
    at Object.connect (node:_tls_wrap:1631:48)
    at Agent.createConnection (node:https:150:22)
    at Agent.createSocket (node:_http_agent:341:26)
    at Agent.addRequest (node:_http_agent:288:10)
    at new ClientRequest (node:_http_client:343:16)
    at request (node:https:360:10)
    at <anonymous> (/Users/user/src/tools/http-proxy-inject-js/node_modules/.pnpm/mockttp@3.7.1/node_modules/mockttp/dist/rules/requests/request-handlers.js:491:31)
    at <anonymous> (/Users/user/src/tools/http-proxy-inject-js/node_modules/.pnpm/mockttp@3.7.1/node_modules/mockttp/dist/rules/requests/request-handlers.js:710:9)
```

As I've figured out from the source code, this is because, somehow, my Node.js version doesn't support `ffdhe2048` and `ffdhe3072` curves. If I comment out those lines in `mockttp` code, the error vanishes:

https://github.com/httptoolkit/mockttp/blob/e769890bea6deb107ecc3c97114a77747fce6ba6/src/rules/passthrough-handling.ts#L31-L35

Node.js version: 18.16.0
`openssl version` output: LibreSSL 2.8.3

From what I've read [here](https://github.com/nodejs/node/issues/39900), it seems like those curves is OpenSSL feature. `mockttp` code currently checks if Node.js version is >= 17 too see if it should try to use them. However, my system has some other OpenSSL version installed

Does anybody know if Node.js relies on the globally installed SSL version? Maybe `mockttp` should provide some option to opt-out from using those new curves? Or maybe I should install OpenSSL 3.0 (and we should add this as note to the documentation)?

**Comment by pimterry** [comment]
Hmm, that's odd. I think I've just found a better approach though which would clean thisup: we can query the actual OpenSSL version being used directly via `process.versions.openssl`, instead of making assumptions from the Node version.

Do you want to open a PR to check against that instead for NEW_CURVES_SUPPORTED?

**Comment by acunniffe** [comment]
+1 We've seen some Optic users (esp in corporate environments w/ pinned openssl versions) run into this as well. 

I think the fix of checking `process.versions.openssl` makes sense and we can help write that if it's helpful.

**Comment by azerum** [comment]
@pimterry Thanks for the idea. From reading the OpenSSL [docs](https://www.openssl.org/policies/releasestrat.html), it seems like they have followed some weird versioning scheme with letters in the end and then switched to more traditional semver. Since FFDHE support was added since 3.0, I guess it's sufficient to set the flag like this?

```
const NEW_CURVES_SUPPORTED = process.versions.openssl.startsWith('3');
```

I am not sure whether `openssl` is guaranteed to be set. If such case is possible, we might assume that curves are not supported if `openssl` is not there:

```
const NEW_CURVES_SUPPORTED = process.versions.openssl?.startsWith('3') ?? false;
```

If everybody agrees with such idea, I can submit a quick PR

**Comment by pimterry** [comment]
Sounds good! It would be mildly better to check `startsWith('3.')` (i.e. with `.`) just to cover our bases and make this a bit more explicit, and we should add a comment to document why we're not just parsing with semver (because old versions aren't actually parseable).

I agree with assuming the curves are not available if the version test fails for some reason.

Happy for either of you (or anybody else) to pick this up.

**Comment by azerum** [comment]
Yeah, idea with `'3.'` is better. But what about possible future major versions of OpenSSL? I guess it's unlikely that they will drop support for new curves? I've also found out that semver can cope with previous versions if `loose: true` option is enabled 

Here's a PR to take a loot at: #140

**Comment by pimterry** [comment]
Released in v3.7.2. Thanks for contributing!
